pkg/webhook/validation/uservalidation.go

```diff
@@ -1,43 +1,71 @@
 package validation
 
 import (
-	"context"
+	"fmt"
 
 	authenticationv1 "k8s.io/api/authentication/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/klog/v2"
 	"k8s.io/utils/strings/slices"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	fleetv1alpha1 "go.goms.io/fleet/apis/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
 const (
-	mastersGroup = "system:masters"
+	mastersGroup         = "system:masters"
+	serviceAccountsGroup = "system:serviceaccounts"
+
+	crdAllowedFormat           = "user: %s in groups: %v is allowed to modify fleet CRD: %+v"
+	crdDeniedFormat            = "user: %s in groups: %v is not allowed to modify fleet CRD: %+v"
+	fleetResourceAllowedFormat = "user: %s in groups: %v is allowed to modify fleet resource %s: %+v"
+	fleetResourceDeniedFormat  = "user: %s in groups: %v is not allowed to modify fleet resource %s: %+v"
 )
 
-// ValidateUserForCRD checks to see if user is authenticated to make a request to modify fleet CRDs.
-func ValidateUserForCRD(whiteListedUsers []string, userInfo authenticationv1.UserInfo) bool {
-	return isMasterGroupUserOrWhiteListedUser(whiteListedUsers, userInfo)
-}
+var (
+	fleetCRDGroups = []string{"networking.fleet.azure.com", "fleet.azure.com", "multicluster.x-k8s.io", "placement.azure.com"}
+)
 
-// ValidateUserForFleetCR checks to see if user is authenticated to make a request to modify Fleet CRs.
-func ValidateUserForFleetCR(ctx context.Context, client client.Client, whiteListedUsers []string, userInfo authenticationv1.UserInfo) bool {
-	if isMasterGroupUserOrWhiteListedUser(whiteListedUsers, userInfo) {
-		return true
+// ValidateUserForFleetCRD checks to see if user is not allowed to modify fleet CRDs.
+func ValidateUserForFleetCRD(group string, namespacedName types.NamespacedName, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
+	if !checkCRDGroup(group) && isMasterGroupUserOrWhiteListedUser(whiteListedUsers, userInfo) {
```

Comment:
Hi Arvind! Should this be an OR term?

Reply:
I've updated the condition to only allow master group users and whitelisted users to modfiy fleet CRDs. We allow any user to modify other CRDs created by the user